integration-tests/custom-environment/lib.rs

```diff
@@ -9,10 +12,11 @@ use ink::env::{DefaultEnvironment, Environment};
 pub enum EnvironmentWithManyTopics {}
 
 impl Environment for EnvironmentWithManyTopics {
-    // We allow for 5 topics in the event, therefore the contract pallet's schedule must
-    // allow for 6 of them (to allow the implicit topic for the event signature).
+    // We allow for 3 topics in the event, including the implicit topic for the event
+    // signature. Therefore, the contract pallet's schedule must also allow for at
+    // least 3 of them.
     const MAX_EVENT_TOPICS: usize =
-        <DefaultEnvironment as Environment>::MAX_EVENT_TOPICS + 1;
+        <DefaultEnvironment as Environment>::MAX_EVENT_TOPICS - 1;
```

Comment:
Wouldn't it be better to change the `MAX_EVENT_TOPICS` in the `DefaultEnvironment`?

Reply:
`MAX_EVENT_TOPICS` has the correct value in `DefaultEnvironment`; it is exactly the same as in the contracts pallet. It already takes into consideration the implicit topic for the event signature, and in both cases, it is set to 4 (3 topics + implicit topic for event signature). To give the test meaning, I set `MAX_EVENT_TOPIC` to 3 (including the event signature). In fact, I am checking if the restriction for 2 topics is applied/checked (2 topics + event signature).